diagnostic_feedback/tests/integration/base_test.py

```diff
@@ -0,0 +1,166 @@
+
+    def _get_final_results(self):
+        return self.browser.find_elements_by_css_selector('div.response_body div.result')
+
+    def load_scenario(self, xml_file, params=None, load_immediately=True, view='student_view'):
+        """
+        Given the name of an XML file in the xml_templates folder, load it into the workbench.
+        """
+        params = params or {}
+        scenario = loader.render_template("xml_templates/{}".format(xml_file), params)
+        self.set_scenario_xml(scenario)
+        if load_immediately:
+            return self.go_to_view(view)
+
+    def load_student_view(self, xml_file, params=None):
+        """
+        Load student view in workbench
+        """
+        self.load_scenario("student_{}.xml".format(xml_file), params=params)
+
+    def load_studio_view(self, xml_file, params=None):
+        """
+        Load studio view in workbench
+        """
+        self.load_scenario("studio_{}.xml".format(xml_file), params=params, view='studio_view')
+
+
+class StudentBaseTest(DiagnosticFeedbackBaseTest):
+
+    def _get_startover_button(self):
+        return self.browser.find_elements_by_xpath("//ul[@role='menu']/li")[3]
+
+    def _get_startover_button_link(self):
+        return self.browser.find_element_by_xpath("//ul[@role='menu']/li/a[@href='#cancel']")
+
+    def _get_visible_choices(self):
+        """
+        Return one OR all visible choices
+        """
+        return [c for c in self.browser.find_elements_by_xpath("//input[@type='radio']") if c.is_displayed()]
+
+    def _is_any_choice_selected(self):
+        return any(choice.is_selected() for choice in self._get_visible_choices())
+
+    def _verify_question(self, question_order=0, next_btn=None, back_btn=None, choice_idx=0):
+        self.assertFalse(self._is_any_choice_selected(), False)
+
+        if question_order > 0:
+            self.assertEqual(back_btn.get_attribute('class'), '')
+        else:
+            self.assertEqual(back_btn.get_attribute('class'), 'disabled')
+
+        self.assertEqual(next_btn.get_attribute('class'), 'disabled')
+
+        choice = self._get_visible_choices()[choice_idx]
+        choice.click()
+        self.assertTrue(self._is_any_choice_selected(), True)
+
+        if question_order == 0:
+            self.assertEqual(back_btn.get_attribute('class'), 'disabled')
+        else:
+            self.assertEqual(back_btn.get_attribute('class'), '')
+
+        self.assertEqual(next_btn.get_attribute('class'), '')
+
+    def _verify_startover(self, next_btn, back_btn, next_btn_link, back_btn_link, startover_link):
+
+        startover_link.click()
+        self.wait_until_visible(next_btn)
+        self.assertFalse(startover_link.is_displayed(), False)
+        self.wait_until_has_class_disabled(next_btn)
+        self.assertEqual(back_btn.get_attribute('class'), 'disabled')
+        self.assertEqual(next_btn.get_attribute('class'), 'disabled')
```

Comment:
@jareerahsan Can you use `wait_until_has_class_disabled` in here? Also, you're already calling that method on `next_btn` in L112 above, so L114 should be safe to remove.